org.wso2.codequalitymatrices/src/test/java/com/wso2/code/quality/metrics/ChangesFinderTest.java

```
@@ -0,0 +1,97 @@
+        expectedDeltedLinesForPatchString1.add(909);
+        expectedDeltedLinesForPatchString1.add(1081);
+        expectedDeltedLinesForPatchString1.add(1082);
+        expectedDeltedLinesForPatchString1.add(1083);
+        expectedDeltedLinesForPatchString1.add(1087);
+        expectedDeltedLinesForPatchString1.add(1094);
+        expectedDeltedLinesForPatchString1.add(1143);
+        expectedDeltedLinesForPatchString1.add(1144);
+        expectedDeltedLinesForPatchString1.add(1145);
+        expectedDeltedLinesForPatchString1.add(1146);
+        expectedDeltedLinesForPatchString1.add(1147);
+        expectedDeltedLinesForPatchString1.add(1148);
+        expectedDeltedLinesForPatchString1.add(1149);
+        expectedDeltedLinesForPatchString1.add(1150);
+        expectedDeltedLinesForPatchString1.add(1151);
+        expectedDeltedLinesForPatchString1.add(1152);
+        expectedDeltedLinesForPatchString1.add(1153);
+        expectedDeltedLinesForPatchString1.add(1165);
+
+        assertThat(actualDeltedLinesForPatchString.size(), is(expectedDeltedLinesForPatchString1.size()));
+        assertThat(actualDeltedLinesForPatchString, is(expectedDeltedLinesForPatchString1));
+
+        patchString = readFile("patchStrings/patchString2.txt");
+        actualDeltedLinesForPatchString = changesFinder.identifyDeletedLines(patchString);
+        assertTrue(actualDeltedLinesForPatchString.isEmpty());
+    }
+
+    /**
+     * This is used to get the contents of the txt file as a single String
+     *
+     * @param path location path to the relevant txt file
+     * @return single String containing all the contents of the given file
+     */
+    private String readFile(String path) {
+        String result = "";
+        ClassLoader classLoader = getClass().getClassLoader();
+        try {
+            result = IOUtils.toString(classLoader.getResourceAsStream(path));
+        } catch (IOException e) {
+            logger.debug(e.getMessage(), e.getCause());
+        }
```

why exception is ignored here. better throw the exception. if we need to ignore, log it as an error.

---

Fixed in 698c71ed0e32960930e00a53d3272028ab2d399a